stdlib/public/core/Sort.swift.gyb

```diff
@@ -53,10 +62,23 @@ func _insertionSort<C>(
       repeat {
         let predecessor: C.Iterator.Element = elements[elements.index(before: i)]
 
-        // if x doesn't belong before y, we've found its position
+        % if p:
+        // If clouser throws the error, We catch the error put the element at right
+        // place and rethrow the error.
+        do {
+          // if x doesn't belong before y, we've found its position
+          if !${cmp("x", "predecessor", p)} {
+            break
+          }
+        } catch {
+          elements[i] = x
```

Comment:
No, If we are mutating a collection with `sort(by:)` than collection has been changed and at this line we are trying to swap the elements.
For eg: if I have array
 var a = [ 5, 4, 3 ,2, 1]
if we throw at  b == 4 than sort(by:) mutate the collection to
[4, 5, 5, 2, 1]
as the [last step](https://github.com/apple/swift/pull/1527/files/e86fb6e14bbf463f6b591042e52e6c50e6130392#diff-68c134d8478193cbe9299610501ff45bR88)
not able to swap the element and we thrown in between. So that is why we are catching the error and throwing it again. 
It just take care of no elements will lost and add.